rollout/ephemeralmetadata.go

```diff
@@ -14,43 +14,46 @@ import (
 func (c *rolloutContext) reconcileEphemeralMetadata() error {
 	ctx := context.TODO()
 	var newMetadata, stableMetadata *v1alpha1.PodTemplateMetadata
-	if c.rollout.Spec.Strategy.Canary != nil {
+
+	switch {
+	case c.rollout.Spec.Strategy.Canary != nil:
 		newMetadata = c.rollout.Spec.Strategy.Canary.CanaryMetadata
 		stableMetadata = c.rollout.Spec.Strategy.Canary.StableMetadata
-	} else if c.rollout.Spec.Strategy.BlueGreen != nil {
+	case c.rollout.Spec.Strategy.BlueGreen != nil:
 		newMetadata = c.rollout.Spec.Strategy.BlueGreen.PreviewMetadata
 		stableMetadata = c.rollout.Spec.Strategy.BlueGreen.ActiveMetadata
-	} else {
+	default:
 		return nil
 	}
+
 	fullyRolledOut := c.rollout.Status.StableRS == "" || c.rollout.Status.StableRS == replicasetutil.GetPodTemplateHash(c.newRS)
+	var err error
 
 	if fullyRolledOut {
-		// We are in a steady-state (fully rolled out). newRS is the stableRS. there is no longer a canary
-		err := c.syncEphemeralMetadata(ctx, c.newRS, stableMetadata)
-		if err != nil {
-			return err
-		}
+		// If the rollout is fully completed, we're in a steady state where newRS is the stableRS and there is no longer a canary.
+		// Sync the stable metadata to the newRS.
+		err = c.syncEphemeralMetadata(ctx, c.newRS, stableMetadata)
 	} else {
-		// we are in a upgrading state. newRS is a canary
-		err := c.syncEphemeralMetadata(ctx, c.newRS, newMetadata)
-		if err != nil {
-			return err
-		}
-		// sync stable metadata to the stable rs
-		err = c.syncEphemeralMetadata(ctx, c.stableRS, stableMetadata)
-		if err != nil {
-			return err
+		// If the rollout is not fully completed, we are in an upgrading state where newRS is a canary.
+		// Sync the new metadata to the newRS.
+		err = c.syncEphemeralMetadata(ctx, c.newRS, newMetadata)
+		// If there was no error syncing the new metadata, sync the stable metadata to the stableRS.
+		if err == nil {
```

Comment:
I think this is less clear. I would keep the early err returns thoughtout the ifs...

Reply:
Truth is, if you looking to tighten it up/ lose some indentation, you could keep the return in the if, and then lose the wrapping else entirely